src/net.cpp

```diff
@@ -2382,17 +2385,32 @@ bool CConnman::BindListenPort(const CService& addrBind, bilingual_str& strError,
 
     // Allow binding if the port is still in TIME_WAIT state after
     // the program was closed and restarted.
-    setsockopt(sock->Get(), SOL_SOCKET, SO_REUSEADDR, (sockopt_arg_type)&nOne, sizeof(int));
+    if (sock->SetSockOpt(SOL_SOCKET, SO_REUSEADDR, (sockopt_arg_type)&nOne, sizeof(int)) == SOCKET_ERROR) {
+        strError =
+            strprintf(Untranslated("Error setting SO_REUSEADDR on socket: %s, continuing anyway"),
+                      NetworkErrorString(WSAGetLastError()));
+        LogPrintf("%s\n", strError.original);
+    }
 
     // some systems don't have IPV6_V6ONLY but are always v6only; others do have the option
     // and enable it by default or not. Try to enable it, if possible.
     if (addrBind.IsIPv6()) {
 #ifdef IPV6_V6ONLY
-        setsockopt(sock->Get(), IPPROTO_IPV6, IPV6_V6ONLY, (sockopt_arg_type)&nOne, sizeof(int));
+        if (sock->SetSockOpt(IPPROTO_IPV6, IPV6_V6ONLY, (sockopt_arg_type)&nOne, sizeof(int)) == SOCKET_ERROR) {
+            strError = strprintf(
+                Untranslated("Error setting IPV6_V6ONLY on socket: %s, continuing anyway"),
+                NetworkErrorString(WSAGetLastError()));
+            LogPrintf("%s\n", strError.original);
+        }
 #endif
 #ifdef WIN32
         int nProtLevel = PROTECTION_LEVEL_UNRESTRICTED;
-        setsockopt(sock->Get(), IPPROTO_IPV6, IPV6_PROTECTION_LEVEL, (const char*)&nProtLevel, sizeof(int));
+        if (sock->SetSockOpt(IPPROTO_IPV6, IPV6_PROTECTION_LEVEL, (const char*)&nProtLevel, sizeof(int)) == SOCKET_ERROR) {
+            strError = strprintf(
+                Untranslated("Error setting IPV6_PROTECTION_LEVEL on socket: %s, continuing anyway"),
+                NetworkErrorString(WSAGetLastError()));
```

Comment:
7c914fe2 These 3 error messages would be easier to read and review if they were on one line like the two pre-existing ones in the same `BindListenPort()` function. Also, the first added one is formatted differently from the next two added ones in this commit. Each being on one line would be simpler to write and to read.

Reply:
Changed to be all in one line, for consistency with the pre-existent ones.

Reply:
Thanks.